clippy_lints/src/methods/mod.rs

```diff
@@ -1052,8 +1052,7 @@ declare_clippy_lint! {
     ///
     /// **Why is this bad?** Readability.
     ///
-    /// **Known problems:** False positive in pattern guards. Will be resolved once
-    /// non-lexical lifetimes are stable.
+    /// **Known problems:** None.
```

Comment:
Can we add tests that assert this?

Reply:
I found the commit where this was added (33847b579eb5479f888a0caae37ebc469dd9ccd2). I'm not what the false positive cases actually look like.